### DISCUSSION TITLE
Blazored modal get initialized again when closing an older one

### DISCUSSION THREAD

**Discussion by shasastf** [Blazored/Modal#470] [Q&A] [answered]
On the main page of my application there's multiple buttons that allow you to create blazored modal (popover) using different methods that look like this:
```
public async Task CreateInstrument()
    {
        if (!instrumentOpening && instrumentGenerator == null)
        {
           instrumentOpening = true;
           InstrumentViewManager instrumentViewManager = new();
            DataAcquisitionAsync.InstrumentData = instrumentViewManager.instrumentData;

            CommunicationCommand command = SendCommandToThreadPool(CommunicationCommandEnum.OpenInstrument);
            bool result = await command.WaitForCompletionAsync();
            if (result)
            {
                ModalOptions options = new() { UseCustomLayout = true };
                ModalParameters parameters = new();

                parameters.Add(nameof(InstrumentMain.InstrumentViewManager), instrumentViewManager);
                parameters.Add(nameof(InstrumentMain.CurrentStyle), CurrentStyle);
                parameters.Add(nameof(InstrumentMain.DeviceName), Name);
                parameters.Add(nameof(InstrumentMain.Module), this);
                parameters.Add(nameof(InstrumentMain.CommunicationThreadPool), communicationThreadPool);

                instrumentsPosition.Add(InstrumentEnum.Instrument);
                ChangeInstrumentsPosition(InstrumentEnum.Instrument);
                instrument = ModalServices.Show<InstrumentMain>("Instrument", parameters, options);
                instrumentOpening = false;
                connectivityTimer.Stop();
                await instrument.Result;

                instrumentsPosition.Remove(InstrumentEnum.Instrument);
                instrument = null;
                if (Connected && !IsInstrumentOpened())
                    connectivityTimer.Start();
            }
        }
    }
```

Every instruments are created correctly, but whenever I close an instrument, all the other that got created after have their `OnInitialized` methods that get called again and all their variables get their base value again.
I know that there has been multiples discussions about thing similar to this, but almost all of them never really got a solution. 
This issue [366](https://github.com/Blazored/Modal/issues/366) seems really similar to my problem, but I wasn't able to make the @key work since my blazored modal aren't created inside a loop.

Is there a way to use the @key or something similar that would prevent this problem?

**Comment by LoupaLoupa** [comment]
I work with [shasastf](https://github.com/shasastf) and we have tried to use update v.7.1.0 where the modification in issue [366](https://github.com/Blazored/Modal/issues/366) should be. The problem still occurs, if we open 2 Modaled instruments (Of different type) in a row and close the First one before the second one, the second one OnInitialized() is called again, and its position is reset because it is local private variables. 

Any hints on how to resolve this issue?

Thanks for any tip you can provide :)

**Comment by LoupaLoupa** [comment, marked as answer]
I have figured a work around, by using a local object in the function that creates the modal and then pass it as a ref parameter, this keeps the position on refresh and stay alive in the context since the Modal keeps the context alive while open. If we call the function again, it creates a new context and we could have a second instrument of the same type if we wanted (We don't right now, we prevent it, but future project will).
